pkg/sharings/sharings.go

```diff
@@ -210,30 +210,53 @@ func ShareDoc(instance *instance.Instance, sharing *Sharing, recStatus *Recipien
 
 		// Dynamic sharing
 		if rule.Selector != "" {
-			// Create index based on selector to retrieve documents to share
-			indexName := "by-" + rule.Selector
-			index := mango.IndexOnFields(docType, indexName, []string{rule.Selector})
-			err := couchdb.DefineIndex(instance, index)
-			if err != nil {
-				return err
-			}
 
-			var docs []couchdb.JSONDoc
+			// Particular case for referenced_by: use the existing view
+			if rule.Selector == "referenced_by" {
+				// The permission tilte is used to know the referenced doctype
+				refType := rule.Title
```

Comment:
Yes I know, I'm not really satisfied by this 'trick', but I didn't see another way to link a reference doctype with the referenced_by selector. We could add a field in the Permissions, but I'm not sure it's worth it.